apps/opik-backend/src/main/java/com/comet/opik/api/AutomationRule.java

```diff
@@ -0,0 +1,39 @@
+package com.comet.opik.api;
+
+import com.fasterxml.jackson.annotation.JsonValue;
+import lombok.AccessLevel;
+import lombok.Getter;
+import lombok.RequiredArgsConstructor;
+
+import java.time.Instant;
+import java.util.Arrays;
+import java.util.UUID;
+
+public sealed interface AutomationRule<T> permits AutomationRuleEvaluator {
```

Comment:
Let's make this interface and its subclasses self de/serialisable without requiring any external help. You just need to configure it with `JsonTypeInfo` and `JsonSubTypes` annotations etc., based on the `action` property.